submissions/andrewklmn/task_js_dom/app.js

```diff
@@ -0,0 +1,209 @@
+      habitat: "Venice, Santa Monica",
+      method: "How to attract her: Don’t shave, use deodorant or have a job, and always carry around a little weed (Don’t worry, it’s practically legal here).",
+    },
+    {
+      menu: 'Gold Digger',
+      header: "The Gold Digger",
+      image: "img/digger.jpg",
+      description: "Sadly, this is one of the more common types of L.A. women, but luckily they’re easy to spot. The twist here is that you don’t necessarily have to have a lot of money, you just have to have the potential to make it. So if you’re a junior agent or work in the mailroom at an entertainment law firm, she’ll invest her time in you. But if you don’t come through, buddy, you are screwed. All she wants is the house on the hill and the fancy car, so if you’re just looking for a trophy wife, then take your pick.",
+      habitat: "Beverly Hills, Hollywood Hills, any Hills, really.",
+      method: "How to attract her: What are you, an idiot? Just flash your cash!",
+    },
+  ],
+  activeIndex: null,
+};
+
+const menuToogleClickListener = (event)=>{
+  let elem = event.target;
+  let aside = document.querySelector('aside');
+  let wraper = document.querySelector('.wraper');
+  
+
+  if (elem.innerHTML=="«") {
+    elem.innerHTML = "☰";
+    elem.style.left = "0";
+    aside.style.width = "0";
+    wraper.style.marginLeft = "20px";
+  } else {
+    elem.innerHTML = "«";
+    elem.style.left = "220px";
+    aside.style.width = "220px";
+    wraper.style.marginLeft = "240px";
+  }
+
+  event.stopPropagation();
+  return false;
+};
+
+const menuItemClickListener = (event)=>{
+  let elem = event.target;
+  state.activeIndex = elem.parentNode.id;
+  renderPage(state);
+
+  const vw = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+  if (vw < 820) {
+    document.querySelector('.btn-toogle-menu').click(); 
+  }
+
+  event.stopPropagation();
+  return false;
+};
+
+const renderMenu = (state)=>{
+  
+  let nav = document.querySelector('nav');
+  let prevUl = document.querySelector('ul');
+  let fragment = document.createDocumentFragment();
+  let ul = document.createElement("ul");
+ 
+ state.items.forEach((item,index)=>{
+    let li = document.createElement("li");   
+    li = document.createElement("li");
+
+    let a = document.createElement("a");
+    li.id = index;
+      
+    a = document.createElement("a");
+    a.innerHTML = item.menu;
+    a.className = "menu-item";
+    a.href = "javascript:void(0)";
+    
+    if (index==state.activeIndex) {
+      a.classList.toggle("active");
+    };
+    
+    a.addEventListener('click',menuItemClickListener);
+    li.appendChild(a);
+    ul.appendChild(li);
+  });
+
+  fragment.appendChild(ul);
+  prevUl.remove();
+  nav.appendChild(fragment);
+};
+
+const showTextContent = ()=>{
+  document.body.style.color = "black";
+}
+
+const renderArticle = (content)=> {
```

Comment:
To my mind, there are too many DOM operations here. How about adding these all content to your HTML and then change only text content from the `state` object. Let me know if you got me. I'm ready for help to discuss it if you'll have some questions.